### DISCUSSION TITLE
Standalone Plugin

### DISCUSSION THREAD

**Discussion by dantleech** [apache/devlake#5251] [Q&A] [answered]
:wave: hi, I'm currently evlauating DevLake, and it looks really good! But our company uses ClickUp to track issues.

There is an open issue to add support [here](https://github.com/apache/incubator-devlake/issues/3458) but in the meantime I'm wondering if it's possible to:

- Write a (lightweight and opinionated) standalone plugin to integrate ClickUp - I see `PLUGIN_DIR` in `env.example` and if it's possible - is there an example anywhere?
- Connect to the DB and insert data via. a parallel, custom, service?

Any advice appreciated!

**Comment by klesh** [comment, marked as answer]
No, not really, data collection is not a one-time deal, the standalone plugin would not work well for our scenario, for example, we don't want to collect all data every time, and collected data must be crunched with others at some point.

It would be much more practical to just folk the repo, copy and modify an existing plugin like `jira`, it is pretty mature at this point.

**Comment by dantleech** [comment]
ok, have forked and made some progress, thanks :+1:

**Reply by DanielGorlo** [reply]
Can you please share? Our company also uses ClickUp and we'd like to use and contribute to the plugin

**Reply by dantleech** [reply]
I no longer work for the company. The fork is [here](https://github.com/altovita) but it's based on a very old version
